src/gpu/generic/sycl/sycl_post_ops.hpp

```diff
@@ -172,6 +184,28 @@ struct sycl_post_ops_t {
     // the number of post ops.
     static constexpr int max_post_ops = 5;
 
+    template <bool allow_inputs = true, bool allow_sum = true>
```

Comment:
Could you make these function arguments instead of template arguments? (Unless there is a good reason for the templatization?)

Reply:
What would be the reason to make them function arguments? I usually try to pass things around as templates if possible for performance reasons (although that should not matter in this case).

Reply:
The main reason is library size. The more we use template argument, the more it makes library size grow. This is even more true as templatization tend to spread to caller site, leading to even more library size growth.

With respect to the performance reasons, I get it for functions called within loops or in the critical path. In this particular case though, this is a post_op check so performance of an extra compare should not matter much.

Reply:
done